clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -322,16 +314,14 @@ private void onErrorResponse(final ConsumerGroupHeartbeatResponse response,
                 break;
 
             case FENCED_MEMBER_EPOCH:
-                message = String.format("GroupHeartbeatRequest failed because member ID %s with epoch %s is invalid. " +
-                                "Will abandon all partitions and rejoin the group",
+                message = String.format("GroupHeartbeatRequest failed because member ID %s with epoch %s is invalid.",
```

Comment:
nit: While we are here, how about `GroupHeartbeatRequest failed because epoch {} is fenced.`?.

Reply:
Sounds good, (just added the member ID too, so `GroupHeartbeatRequest failed for member %s because epoch %s is fenced.`)